### DISCUSSION TITLE
Scaling of crystal shape plot in Matlab Figure

### DISCUSSION THREAD

**Discussion by LCedrig** [mtex-toolbox/mtex#2236] [Ask Anything] [answered]
Dear all, 

the plotting of crystal shapes is great (also on its own). However, I can not seem to ensure that the scaling of the plotted crystal shape always fits into the figure. This can also be seen on some of the plots in the MTEX documentation of crystal shapes (https://mtex-toolbox.github.io/CrystalShapes.html).

See for example this plot.
![image](https://github.com/user-attachments/assets/6b9c67ed-8c87-4aba-b804-3ef4ce992270)

How can I make the crystal shape fit into the Matlab figure? ( I am using mtex 5.11.2)
Thank you!

**Comment by zmichels** [comment]
In the documentation you linked, it is mentioned how to scale the size of the crystalShapes:

- factor * cS scales the crystal shape in size
- ori * cS rotates the crystal shape in the defined orientation
- [xy] + cS or [xyz] + cS shifts the crystal shape in the specified positions

**Comment by LCedrig** [comment]
Thank you very much for your suggestion. I had tried the scaling.

The following code (slightly adapted from the documentation)  provides a plot inside the figure that appears cropped at the top and bottom inside the figure. This behavior, i.e. the graphical representation of the crystal shape inside the figure object, is independent of the scaling argument `scale` (I am not an experienced Matlab user). 

My main interest is to export the nice graphical visualization of the crystal shape as a vector graphic. 


```

cs = loadCIF('quartz')

m = Miller({1,0,-1,0},cs);  % hexagonal prism
r = Miller({1,0,-1,1},cs);  % positive rhomboedron, usally bigger then 
z = Miller({0,1,-1,1},cs);  % negative rhomboedron
s1 = Miller({2,-1,-1,1},cs);% left tridiagonal bipyramid
s2 = Miller({1,1,-2,1},cs); % right tridiagonal bipyramid
x1 = Miller({6,-1,-5,1},cs);% left positive Trapezohedron
x2 = Miller({5,1,-6,1},cs); % right positive Trapezohedron

N = [2.5*m,r,0.9*z,0.7*s1,0.3*x1];
cS = crystalShape(N)
scale = 0.5
plot(scale*cS,'colored')
```

**Comment by zmichels** [comment, marked as answer]
Yah, I see now... even in the documentation, some of the figures are clipped out of frame.

add a zoom specification following the plot command... to zoom out and see the whole shape:

```matlab
plot(cS,'colored')
zoom(.5)
```

Also... FWIW, there is a fancy quartz crystal shape that ships with mtex (and for some odd reason, it does not require the zoom to show the whole thing):

```matlab
csq = crystalShape.quartz
plot(csq, 'colored')
```

**Reply by LCedrig** [reply]
Thank you!
Adding the zoom function does the trick.
